compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -889,8 +889,9 @@ object PatternMatcher {
        */
 
       val (primScrutinee, scrutineeTpe) =
-        if (scrutinee.tpe.widen.isRef(defn.IntClass)) (scrutinee, defn.IntType)
-        else if (scrutinee.tpe.widen.isRef(defn.StringClass)) (scrutinee, defn.StringType)
+        val tpe = scrutinee.tpe.widen.widenSingletons()
+        if (tpe.isRef(defn.IntClass)) (scrutinee, defn.IntType)
```

Comment:
@sjrd : Do you see any reason to not use `scrutinee.tpe <:< defn.IntType` here ?

Reply:
No, there is no reason not to. We should definitely use `scrutinee.tpe <:< defn.IntType` and `scrutinee.tpe <:< defn.StringType`, instead of selectively widening some things.